### PR TITLE
Added compatibility for same time delta nodes

### PR DESCRIPTION
**Pull request by dclark87**
Minor fix to allow multiple nodes with the same timestamp (start or finish) to be plotted in the Gantt chart without raising an exception

### CODE REVIEW

**Comment by chrisfilo** [comment]
BTW could you add a smoke test for draw_resource_bar()? Maybe by switching one of the workflows to using MultiProc https://github.com/nipy/nipype/blob/master/circle.yml#L67 and then generate a gantt plot for them (and preferably keep it in artefacts so we could eyeball it).

**Comment by dclark87** [comment]
Ok, I added what little I know about the smoke tests for the draw gantt chart functionality

**Comment by chrisfilo** [comment]
Awesome! This is exactly what we needed!

**Comment by satra** [comment]
@dclark87 - could you please merge with current master and push?

**Comment by dclark87** [comment]
Added some tolerance to the runtime profiler tests and re-based with nipy's master

**Comment by satra** [comment]
@dclark87 - we can't even look at the output log on circle because the workflow debug statements happen every couple of ms. can we reduce the debug statements?

```
160617-22:21:59,543 workflow DEBUG:
	 Free memory (GB): 51, Free processors: 2
```

**Comment by dclark87** [comment]
Hey I made some fixes here to check for pandas (which is why the CI was failing) for the smoke tests. I also re-based with lastest nipy-master. Does this look all good to merge?

File: circle.yml [hunk 1]
```diff
@@ -23,6 +23,28 @@ test:
         timeout: 1600
     - docker run -i -v /etc/localtime:/etc/localtime:ro -v ~/scratch:/scratch -w /scratch nipype/testbench test_spm Linear /root/examples/ workflow4d :
         timeout: 1600
+<<<<<<< HEAD
```
Comment (satra) [inline circle.yml]: this file contains git differences. should be resolved and re-committed

**Comment by dclark87** [comment]
ah crap - will fix those, thanks

**Comment by dclark87** [comment]
@satra - this all g2g?

File: nipype/pipeline/plugins/multiproc.py [hunk 2]
```diff
@@ -219,9 +219,9 @@ def _send_procs_to_workers(self, updatehash=False, graph=None):
         jobids = sorted(jobids,
                         key=lambda item: (self.procs[item]._interface.estimated_memory_gb,
                                           self.procs[item]._interface.num_threads))
-
-        logger.debug('Free memory (GB): %d, Free processors: %d',
-                     free_memory_gb, free_processors)
+        if len(jobids) > 0:
```
Comment (oesteban) [inline nipype/pipeline/plugins/multiproc.py]: I think @satra protected this trace printing with the ```runtime_profile``` boolean as well.

File: nipype/pipeline/plugins/base.py [hunk 3]
```diff
@@ -262,7 +262,6 @@ def run(self, graph, config, updatehash=False):
             if toappend:
                 self.pending_tasks.extend(toappend)
             num_jobs = len(self.pending_tasks)
-            logger.debug('Number of pending tasks: %d' % num_jobs)
```
Comment (oesteban) [inline nipype/pipeline/plugins/base.py]: This trace appeared too often, maybe it's ok to remove it. But I ask @satra first.

File: nipype/interfaces/base.py [hunk 4]
```diff
@@ -1333,6 +1333,9 @@ def get_max_resources_used(pid, mem_mb, num_threads, pyfunc=False):
     try:
         mem_mb = max(mem_mb, _get_ram_mb(pid, pyfunc=pyfunc))
         num_threads = max(num_threads, _get_num_threads(psutil.Process(pid)))
+    except psutil.AccessDenied as exc:
```
Comment (oesteban) [inline nipype/interfaces/base.py]: Maybe disable the ```runtime_profile``` here?

**Comment by oesteban** [comment]
Hi @dclark87, can I ask the reason to close this PR? I think it is really valuable.
If it is something derived from my comments, please accept my apologies.

**Comment by dclark87** [comment]
Hey @oesteban - I closed the PR because I am no longer actively developing or pushing to the FCP-INDI/nipype repo - so I am not able to rebase and make those changes. The maintainers there can open the pull request back up, rebase and address your comments.

**Comment by oesteban** [comment]
Cool thanks!

**Pull request closed without merging**